### PR TITLE
Introduces v1alpha4 types

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
This PR generates the types for the v1alpha4 API for CAPV
Tasks:
- [x] Generate v1alpha4 types
- [x] Remove v1alpha2 types
- [x] Mark v1alpha4 as the conversion hub

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes ##1038

**Special notes for your reviewer**:
Yet to remove the deprecated types, will create another PR for that purpose

**Release note**:
```release-note
Generated v1alpha4 API version types and removed v1alpha2 types
```

### CODE REVIEW

**Review by yastij — COMMENTED** [review]
/approve
/lgtm
/hold

@srm09 - just a few nits, feel free to remove the hold If you plan on opening a PR to remove the deprecated APIs

File: api/v1alpha3/haproxyloadbalancer_conversion.go [hunk 1]
```diff
@@ -16,8 +16,31 @@ limitations under the License.
 
 package v1alpha3
 
-// Hub marks HAProxyLoadBalancer as a conversion hub.
-func (*HAProxyLoadBalancer) Hub() {}
+import (
+	infrav1alpha4 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha4"
```
[the remark this answers is not in the document]
Comment (srm09) [reply api/v1alpha3/haproxyloadbalancer_conversion.go:20]: `infrav1` helps in general scenarios wherein we import either v1a3 or v1a4. This I feel is a special case wherein we are importing v1a4 into the v1a3 package.

File: api/v1alpha3/conversion_test.go [hunk 2]
```diff
@@ -14,25 +14,30 @@ See the License for the specific language governing permissions and
...
 	"testing"
 
-	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
-
 	. "github.com/onsi/gomega"
+
 	"k8s.io/apimachinery/pkg/runtime"
-	"sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha3"
+	nextver "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha4"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
```
[the remark this answers is not in the document]
Comment (srm09) [reply api/v1alpha3/conversion_test.go:26]: Done, pushing it

**Comment by srm09** [comment]
@yastij Re-arranged and cleaned the commits to make it ready for merge. I will remove the hold and create another PR for removing the deprecated stuff.
/unhold

**Pull request merged**